cmd/ncproxy/networkagent/main.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+
+	"github.com/Microsoft/hcsshim/cmd/ncproxy/configagent"
+	"github.com/Microsoft/hcsshim/cmd/ncproxy/ncproxygrpc"
+	"github.com/Microsoft/hcsshim/internal/hcsoci"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+// This is a barebones example of an implementation of the network
+// config agent service that ncproxy talks to. This is solely used to test and
+// will be removed.
+
+const (
+	netID       = "ContainerPlat-nat"
+	listenAddr  = "127.0.0.1:9201"
+	ncProxyAddr = "127.0.0.1:9200"
+)
+
+type service struct {
+	client ncproxygrpc.NetworkConfigProxyClient
+}
+
+func (s *service) ConnectNamespaceToNetwork(ctx context.Context, req *configagent.ConnectNamespaceToNetworkRequest) (*configagent.ConnectNamespaceToNetworkResponse, error) {
+	// Change NetworkID to NetworkName if this is the preferred method? Easier to
+	// debug etc.
```

Comment:
?

Reply:
We currently have the design to pass in the HNS network ID which isn't as informative in logs and a bigger pain to find, and as of now we pass in the network name into this field even though it's called NetworkID. I just kept this here as a reminder to swap over if on validation of this it's decided that that's the preferred way. This won't be staying once that's decided 😄